src/middlewared/middlewared/plugins/vm.py

```diff
@@ -331,13 +331,19 @@ def __init__(self, manager, vm):
         vm = await self.middleware.call('datastore.query', 'vm.vm', [('id', '=', id)])
         guest_memory = vm[0].get('memory', 0) * 1024 * 1024
         arc_max = sysctl.filter('vfs.zfs.arc_max')[0].value
+        arc_min = sysctl.filter('vfs.zfs.arc_min')[0].value
         new_arc_max = min(
             await self.middleware.call('vm.get_initial_arc_max'),
             arc_max + guest_memory
         )
         if arc_max != new_arc_max:
-            self.logger.debug(f'===> Give back guest memory to ARC: {new_arc_max - arc_max}')
-            sysctl.filter('vfs.zfs.arc_max')[0].value = new_arc_max
+            if new_arc_max > arc_min:
+                self.logger.debug(f'===> Give back guest memory to ARC: {new_arc_max}')
+                sysctl.filter('vfs.zfs.arc_max')[0].value = new_arc_max
+            else:
+                self.logger.warn(
+                    f'===> Not giving back memory to ARC because new arc_max ({new_arc_max}) <= arc_min ({arc_min})'
+                )
```

Comment:
Shouldnt we make sure arc_max is set to at least arc_min instead?

Reply:
@william-gr I think we are doing exactly it, aren't we?

Reply:
Are we?

I am trying to figure out how new_arc_max could possibly be lower than arc_min?

It seems somehow vm.get_initial_arc_max is getting a low value and thats the root cause of the issue?

Reply:
@william-gr
```
if new_arc_max > arc_min:
    ...
else:
    # new_arc_max <= arc_min
    # "arc_max is set to at least arc_min" is wrong, we are failing
```

Or are you just proposing to change `>` to `>=`?

No idea too why this happens too, my only guess was tunable being changed to lower value while VM was running.

Reply:
I am saying at this point we have no idea what arc_max is currently set to since a guest was teared down and the new values do not make sense. I am pondering if we should set some value in the `else`. Either arc_mix, arc_min + guest_memory, something else or nothing.

Reply:
I think no-op is the best solution:
 * decreasing it to arc_min has no sense at all: VM has stopped, we've got more free memory, why reduce ARC size?
 * increasing it to `arc_min + guest_memory` may be too much: other guests might start swapping

Reply:
You also have no idea what arc_max is. Perhaps it should be investigated why that happened.

Reply:
> You also have no idea what arc_max is

Why do you think so?

> Perhaps it should be investigated why that happened.

How? I've only managed to repeat it as described above: by manually setting sysctl while running VM

Reply:
Its clear we have no idea what new arc_max should be is if its lower then arc_min, no?

Forget it, I will handle it :)